### ISSUE TITLE
Fails to build on Ubuntu (18.10)

### ISSUE THREAD

**Issue by alexmyczko** [pushover/pushover.github.io#13]
Here's a build log: https://launchpadlibrarian.net/369018496/buildlog_ubuntu-cosmic-amd64.pushover_0.0.5+git20180420-2_BUILDING.txt.gz

However in Debian it's fine, and people install it:
https://qa.debian.org/popcon.php?package=pushover

**Comment by vog** [comment]
Are you able to get the output of the following commands on the build server system?

```
pkg-config --cflags SDL_image libpng sdl
pkg-config --libs SDL_image libpng sdl
```

Also, please try again with the current `master`, as the build has been radically simplified (and accelerated) by #10.

**Comment by alexmyczko** [comment]
Yes they say:

`# pkg-config --cflags SDL_image libpng sdl`

-D_GNU_SOURCE=1 -D_REENTRANT -I/usr/include/SDL -I/usr/include/libpng16 -I/usr/include/SDL

`# pkg-config --libs SDL_image libpng sdl`

-lSDL_image -lpng16 -lz -lSDL

ah no, that's where it works...

where it doesn't work they say:

`# pkg-config --cflags SDL_image libpng sdl`
Package SDL_image was not found in the pkg-config search path.
Perhaps you should add the directory containing `SDL_image.pc'
to the PKG_CONFIG_PATH environment variable
No package 'SDL_image' found
`# pkg-config --libs SDL_image libpng sdl`
Package SDL_image was not found in the pkg-config search path.
Perhaps you should add the directory containing `SDL_image.pc'
to the PKG_CONFIG_PATH environment variable
No package 'SDL_image' found

but
`# dpkg -l |grep sdl |grep dev # says`
ii  libsdl2-dev:amd64                             2.0.8+dfsg1-1ubuntu1                        amd64        Simple DirectMedia Layer development files
ii  libsdl2-image-dev:amd64                       2.0.3+dfsg1-1                               amd64        Image loading library for Simple DirectMedia Layer 2, development files

**Comment by vog** [comment]
Okay, so they use the wrong SDL library. They provide SDL-2, but we need SDL-1.2.

(We should move to SDL-2 some day, see #20, but that day is not today.)

More precisely, the following Debian/Ubuntu packages of SDL should be used for building:

* libsdl1.2-dev
* libsdl-image1.2-dev
* libsdl-mixer1.2-dev
* libsdl-ttf2.0-dev

Does this fix the issue?

**Comment by alexmyczko** [comment]
No, they are all installed. Still it'd fail with
make[1]: Leaving directory '/home/alex/pu/pushover-0.0.5+git20180512'
dh_auto_build: make -j8 -O "INSTALL=install --strip-program=true" returned exit code 2
debian/rules:12: recipe for target 'build' failed
make: *** [build] Error 25

**Comment by vog** [comment]
> No, they are all installed

What exactly do you mean by that?

As the reported `pkg-config` output was "Package SDL_image was not found", it is pretty certain that the package `libsdl1.2-dev` was not installed. Also, the reported output of `dpkg -l | grep sdl | grep dev` did not show any of the SDL-1.2 packages, just SDL-2 packages.

> make[1]: Leaving directory '/home/alex/pu/pushover-0.0.5+git20180512'
> dh_auto_build: make -j8 -O "INSTALL=install --strip-program=true" returned exit code 2
> debian/rules:12: recipe for target 'build' failed
> make: *** [build] Error 25

Would you mind providing the full build log?

**Comment by vog** [comment]
@alexmyczko Any chance to get a full build log?

**Comment by alexmyczko** [comment]
Here's a full build log.

[pushover_0.0.5+git20180420-2_amd64.build.txt](https://github.com/pushover/pushover.github.io/files/2076672/pushover_0.0.5.git20180420-2_amd64.build.txt)

**Comment by vog** [comment]
That one seems to build an old version of pushover.

Could you rebuild with the current version, please? That is, commit d5b6816865853697fcc664bf4be2ca887e63dfb3 from 2018-05-11.

**Comment by alexmyczko** [comment]
Ok I'll try a latest checkout soon...

**Comment by vog** [comment]
Thanks!

**Comment by vog** [comment]
Any progress on this?

Otherwise, I'd propose we make the release first, and take care of this strange build issue later.

**Comment by alexmyczko** [comment]
nope, i am offline two weeks, go ahead...

**Comment by alexmyczko** [comment]
There's the patch/fix http://launchpadlibrarian.net/384905316/pushover_0.0.5+git20180420-2_0.0.5+git20180420-2ubuntu1.diff.gz

**Comment by vog** [comment]
Fixed by #21.
